treeherder/intermittents_commenter/commenter.py

```diff
@@ -297,46 +309,94 @@ def get_bug_stats(self, startday, endday):
             .filter(total__gte=threshold)
             .values_list("bug_id", flat=True)
         )
-
         bugs = (
             BugJobMap.failures.by_date(startday, endday)
             .filter(bug_id__in=bug_ids)
+            .order_by("job__machine_platform__platform")
             .values(
                 "job__repository__name",
                 "job__machine_platform__platform",
                 "bug_id",
                 "job__option_collection_hash",
+                "job__signature__job_type_name",
+                "job__signature__job_group_name",
             )
         )
-
         option_collection_map = OptionCollection.objects.get_option_collection_map()
-        bug_map = dict()
+        bug_map = self.build_bug_map(bugs, option_collection_map)
+        return bug_map, bug_ids
 
+    def fetch_test_variants(self):
+        mozilla_central_url = "https://hg.mozilla.org/mozilla-central"
+        variant_file_url = f"{mozilla_central_url}/raw-file/tip/taskcluster/kinds/test/variants.yml"
+        response = requests.get(variant_file_url, headers={"User-agent": "mach-test-info/1.0"})
+        return yaml.safe_load(response.text)
+
+    def get_test_variant(self, test_suite):
+        test_variants = self.fetch_test_variants()
+        # iterate through variants, allow for Base-[variant_list]
+        variant_symbols = sorted(
+            [
+                test_variants[v]["suffix"]
+                for v in test_variants
+                if test_variants[v].get("suffix", "")
+            ],
+            key=len,
+            reverse=True,
+        )
+        # strip known variants
+        # build a list of known variants
+        base_symbol = test_suite
+        for variant in variant_symbols:
+            if f"-{variant}" in base_symbol:
+                return variant
+        return "no variant"
+
+    def build_bug_map(self, bugs, option_collection_map):
+        """Returning a dict of bug_id's with total, repository, test_suite and platform totals"""
+        bug_map = dict()
         for bug in bugs:
             platform = bug["job__machine_platform__platform"]
             repo = bug["job__repository__name"]
             bug_id = bug["bug_id"]
             build_type = option_collection_map.get(
                 bug["job__option_collection_hash"], "unknown build"
             )
-
-            if bug_id in bug_map:
-                bug_map[bug_id]["total"] += 1
-                bug_map[bug_id]["per_repository"][repo] += 1
-                bug_map[bug_id]["per_platform"][platform] += 1
-                if bug_map[bug_id].get(platform):
-                    bug_map[bug_id][platform][build_type] += 1
-                else:
-                    bug_map[bug_id][platform] = Counter([build_type])
-
+            test_variant = self.get_test_variant(bug["job__signature__job_type_name"])
+            test_group_name = bug["job__signature__job_group_name"]
+            test_variant = f"{test_group_name} / {test_variant}"
```

Comment:
I think we don't need to include test_group_name, just leave test_variant as it is